File: homework/HW11/PythonVersion/FloodFill.py
class colorFlood:
    def floodFill(self, image, sr, sc, color):
        if(image[sr][sc] == color):
            return image
        self.fill(self,image,sr,sc,image[sr][sc], color)
        return image

    def fill(self, image, sr, sc, color, initialColor):
        if(sr < 0 or sc < 0 or sr >= len(image) or sc >= len(image[0]) or image[sr][sc] != color):
            return
        image[sr][sc] = initialColor

        self.fill(self,image, sr-1, sc, color, initialColor)
        self.fill(self,image, sr+1, sc, color, initialColor)
        self.fill(self,image, sr, sc-1, color, initialColor)
        self.fill(self,image, sr, sc+1, color, initialColor)


x = colorFlood
test = [[1,1,1], [1,1,0], [1,0,1]]
test1 = [[0,0,0],[0,0,0]]

print("Test Case 1:")
print("Initial: ")
for i in range(len(test)):
    for j in range(len(test[0])):
        print("[" + str(test[i][j]) + "] ", end= "")
    print()

result = x.floodFill(x,test,1,1,2)

print("\nAfter Flood Fill:")
for i in range(len(test)):
    for j in range(len(test[0])):
        print("[" + str(result[i][j]) + "] ", end= "")
    print()

print("\nTest Case 2:")
print("Initial: ")

for i in range(len(test1)):
    for j in range(len(test1[0])):
        print("[" + str(test1[i][j]) + "] ", end= "")
    print()
result = x.floodFill(x,test1,0,0,0)

print("\nAfter Flood Fill:")
for i in range(len(test1)):
    for j in range(len(test1[0])):
        print("[" + str(result[i][j]) + "] ", end= "")
    print()

File: homework/HW11/JavaVersion/FloodFill.java
package homework.HW11.JavaVersion;

public class FloodFill {
    
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        
        //Base Case, if starting pixel equals color, no changes are made
        if(image[sr][sc] == color){
            return image;
        }
        fill(image, sr, sc, image[sr][sc], color);      //Call recursive fill function
        return image;
    }

    public void fill(int[][]image, int sr, int sc, int color, int initialColor){
        //Base Case, if current position is out of bounds or the color is not equal to the initial color
        if(sr < 0 || sc < 0 || sr >= image.length || sc >= image[0].length || image[sr][sc]  != color){
            return;
        }
        image[sr][sc] = initialColor;       //Change pixel color to initial Color

        //Recursive calling 4-directionally pixels
        fill(image, sr - 1, sc, color, initialColor);
        fill(image, sr + 1, sc, color, initialColor);
        fill(image, sr, sc - 1, color, initialColor);
        fill(image, sr, sc + 1, color, initialColor);

    }

    public static void main(String[] args) {

        FloodFill x = new FloodFill();

        
        int[][]test = {{1,1,1},{1,1,0},{1,0,1}};
        int[][]test1 = {{0,0,0},{0,0,0},};

        //First test case
        System.out.println("Test Case 1:");
        System.out.println("Initial: ");
        for(int i = 0; i < test.length; i++){
            for(int j = 0; j < test[0].length; j++){
                System.out.print("[" + test[i][j] + "] ");
            }
            System.out.println();
        }

        int[][] result = x.floodFill(test, 1, 1, 2);

        System.out.println("\nAfter Flood Fill: ");
        for(int i = 0; i < test1.length; i++){
            for(int j = 0; j < test1[0].length; j++){
                System.out.print("[" + result[i][j] + "] ");
            }
            System.out.println();
        }

        //Second test case
        System.out.println("\nTest Case 2:");

        System.out.println("Initial: ");
        for(int i = 0; i < test1.length; i++){
            for(int j = 0; j < test1[0].length; j++){
                System.out.print("[" + test1[i][j] + "] ");
            }
            System.out.println();
        }

        result = x.floodFill(test1, 0, 0, 0);

        System.out.println("\nAfter Flood Fill: ");
        for(int i = 0; i < test1.length; i++){
            for(int j = 0; j < test1[0].length; j++){
                System.out.print("[" + result[i][j] + "] ");
            }
            System.out.println();
        }




    }

}
